pkgs/development/compilers/ghc/head.nix

```diff
@@ -15,16 +15,26 @@
 
 , # If enabled, GHC will be built with the GPL-free but slower integer-simple
   # library instead of the faster but GPLed integer-gmp library.
-  enableIntegerSimple ? false, gmp ? null
+  enableIntegerSimple ? targetPlatform != hostPlatform, gmp ? null
 
 , # If enabled, use -fPIC when compiling static libs.
   enableRelocatedStaticLibs ? targetPlatform != hostPlatform
 
 , # Whether to build dynamic libs for the standard library (on the target
   # platform). Static libs are always built.
-  enableShared ? true
+  #
+  # We don't do this by default if target != host, or if we target windows.
+  enableShared ? (targetPlatform == hostPlatform && !targetPlatform.isWindows)
```

Comment:
I believe that the description of this option is actually quite incorrect and misleading. Setting this to true has no effect on whether ghc builds shared libraries or not. Rather, it affects the code that's used by `ghci` to load compiled objects for interpretation. There's a good chance that this setting is completely ignored on Windows anyway. Could you please check whether the change you've made here is actually necessary?

Reply:
`enable*` is about how the *current* package is linked. I think this does just that. That this also affects how GHCi works is just a side affect due to GHCi's fragile interpretation.